plugins/BEdita/Core/src/Model/Table/RolesTable.php

```diff
@@ -124,4 +127,19 @@ protected function findMine(Query $query)
                 return $exp->in($this->aliasField((string)$this->getPrimaryKey()), $subQuery);
             });
     }
+
+    /**
+     * Before delete checks: if record is not deletable, raise a ImmutableResourceException
+     *
+     * @param \Cake\Event\Event $event The beforeSave event that was fired
+     * @param \Cake\Datasource\EntityInterface $entity the entity that is going to be saved
+     * @return void
+     * @throws \BEdita\Core\Exception\ImmutableResourceException if entity is not deletable
+     */
+    public function beforeDelete(Event $event, EntityInterface $entity)
+    {
+        if (static::ADMIN_ROLE === $entity->id) {
+            throw new ImmutableResourceException(__d('bedita', 'Could not delete "{0}" {1}', $entity, $entity->id));
```

Comment:
`$entity` is an object. Is it stringified correctly? 🤔

Reply:
Now we don't need dynamic value... In `RolesTable` entity is `Role`.